### DISCUSSION TITLE
how can we do element transition with javascript hooks, like in case of using gsap?

### DISCUSSION THREAD

**Discussion by jonmes** [nuxt/nuxt#17020] [Questions]
how can we do it with javascript hooks, like in case of using gsap.
I tried it but the hooks are not triggered without the condition or (v-if), it should work without v-if when element is mounted for the first time, I tried the same exact code on vue 3 and it works but not with nuxt 3.

code is here for Nuxt 3
```
<script setup>
import gsap from "gsap";

const images = [
  "https://cdn.pixabay.com/photo/2022/06/02/19/25/giraffes-7238815_960_720.jpg",
  "https://cdn.pixabay.com/photo/2022/06/02/19/25/giraffes-7238815_960_720.jpg",
  "https://cdn.pixabay.com/photo/2022/06/02/19/25/giraffes-7238815_960_720.jpg",
  "https://cdn.pixabay.com/photo/2022/06/02/19/25/giraffes-7238815_960_720.jpg",
];

const beforeEnter = (el) => {
  console.log("beforeEnter ,");
  el.style.transform = "translateY(100px)";
  el.style.opacity = 0;
};

const enter = (el, done) => {
  console.log("enter");
  gsap.to(el, {
    y: 0,
    opacity: 1,
    duration: 0.7,
    onComplete: done,
    delay: el.dataset.index * 0.1,
  });
};
</script>

<template>
  <div class="tran">
    <transition-group
      appear="true"
      @before-enter="beforeEnter"
      @enter="enter"
      name="stagger"
      tag="ul"
    >
      <li v-for="(img, i) in images" class="list" :key="i" :data-index="i">
        <img alt="Vue logo" :src="img" style="width: 100px" />
      </li>
    </transition-group>
  </div>
</template>
```

**Comment by rhernandog** [comment]
The problem seems to be the `appear` attribute in both `<transition>` and `<transition-group>`. I'm using GSAP in a simple Nuxt3 project (RC8) and the page transition doesn't work when the app first runs. If you navigate to different pages, the page transitions work correctly. Same with components or elements inside a page. When you navigate to the page nothing happens, no animations. If you change something on the code, the HMR runs and the animation do happens.
